### DISCUSSION TITLE
Is libcaf_python gone?

### DISCUSSION THREAD

**Discussion by alatdneg** [actor-framework/actor-framework#1212] [Q&A]
Hi, I'm trying to implement a python module , using libcaf_python, but I've just noticed that it's now gone, and atom_value is also gone.

I updated my build off master a few weeks ago, and only just noticed the changes.

Is it still possible to do this ? And if so what should I replace atom_value with ?

Cheers.

**Comment by alatdneg** [comment]
I've almost got it working, but the atoms are causing me grief, I can't work out how to set them up so they are recognised by python and the c++ code.

**Comment by Neverlord** [comment]
We recently dropped `libcaf_python`: https://github.com/actor-framework/actor-framework/issues/775#issuecomment-608623572. That's not to say that we wouldn't add Python bindings in the future again, just that the Python shell didn't had a purpose any more at the moment.

> what should I replace atom_value with ?

CAF 0.18 is going to implement atoms using tag types. The simplest way to define them is by using the `CAF_ADD_ATOM` macro. As an example of the new API, you can have a look at the [dining philosophers](https://github.com/actor-framework/actor-framework/blob/master/examples/dynamic_behavior/dining_philosophers.cpp).

As to how to make binding to the new atoms: either one creates one Python type per atom or one could create a dictionary to look up types by name at runtime.

We will also update the docs if CAF 0.18 nears completion.

**Comment by alatdneg** [comment]
Finally got it working, didn't realise you now need to register each "atom" as a message type as well, couldn't workout why the remote wasn't receiving any messages.

cheers
